subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNode.java

```diff
@@ -226,7 +220,9 @@ public void visitLocalStateProperty(final Object value) {
                     withDeadlockHandling(
                         taskNode,
                         "a local state", "local state properties",
-                        () -> mutations.outputPaths.addAll(canonicalizedPaths(canonicalizedFileCache, fileCollectionFactory.resolving(value))));
+                        () -> fileCollectionFactory.resolving(value)
+                            .forEach(file -> mutations.outputPaths.add(file.getAbsolutePath()))
```

Comment:
Not in this PR, but I wonder if we could use the output visitor in the `UnitOfWork` here somehow eventually...

Reply:
Yeah. Though we don't have the unit of work at this point in time. If we do this kind of parallel avoidance for all nodes in the graph, this will also have a performance impact.